torch/onnx/_internal/fx/exporter.py

```diff
@@ -1058,92 +1059,64 @@ def save_model_with_external_data(
-    torch.uint8: onnx.TensorProto.UINT8,
-    torch.int8: onnx.TensorProto.INT8,
-    torch.int16: onnx.TensorProto.INT16,
-    torch.int32: onnx.TensorProto.INT32,
-    torch.int64: onnx.TensorProto.INT64,
-    torch.float16: onnx.TensorProto.FLOAT16,
-    torch.float32: onnx.TensorProto.FLOAT,
-    torch.float64: onnx.TensorProto.DOUBLE,
-    torch.complex64: onnx.TensorProto.COMPLEX64,
-    torch.complex128: onnx.TensorProto.COMPLEX128,
-    torch.bfloat16: onnx.TensorProto.BFLOAT16,
-}
-
-# TODO(titaiwang): copied from ops_correctness_test.py, should have a common place?
-def _convert_tensor_to_numpy(input: Any) -> Any:
-    if isinstance(input, torch.Tensor):
-        return input.detach().cpu().numpy()
-    if isinstance(input, (tuple, list)):
-        if len(input) == 0:
-            return np.array((), dtype=np.int64)
-        if isinstance(input[0], torch.Tensor):
-            return [_convert_tensor_to_numpy(x) for x in input]
-        if isinstance(input[0], bool):
-            return np.array(input, dtype=np.bool_)
-
-        # Just a sequence of numbers
-        if isinstance(input[0], int):
-            return np.array(input, dtype=np.int64)
-        if isinstance(input[0], float):
-            return np.array(input)
-
-    return input
-
-
-# TODO(titaiwang): copied from ops_correctness_test.py, should have a common place?
-def _convert_kwargs_for_onnx(kwargs: dict[str, Any]) -> dict[str, Any]:
-    """Converts kwargs to be compatible with ONNX Runtime.
-
-    ONNX Runtime doesn't support torch.bool, so we convert them to torch.uint8.
-    """
-    new_kwargs = {}
-    for key, value in kwargs.items():
-        if key == "device":
-            continue
-        if key == "dtype":
-            value = TORCH_TYPE_TO_ONNX[value]
-        new_kwargs[key] = value
-    return new_kwargs
-
-
 @_beartype.beartype
 def _validate_op_between_ort_torch(
     node: torch.fx.Node,
-    symbolic_fn: onnxscript.OnnxFunction,
+    symbolic_fn: Union[onnxscript.OnnxFunction, Callable],
     torch_args: tuple,
     torch_kwargs: dict,
 ):
     """Validate the op between ONNX Runtime and PyTorch."""
     # op-level validation
     # Symbolic_fn should have the same output as node.target (torch ops)
+    # TODO: torch.dtype to onnx.dtype conversion
+    # trace_only function is regular python function
+    function_name = (
+        symbolic_fn.name
+        if isinstance(symbolic_fn, onnxscript.OnnxFunction)
+        else symbolic_fn.__name__
+    )
     try:
         with evaluator.default_as(evaluator.ort_evaluator):
             expected_outputs = node.target(*torch_args, **torch_kwargs)  # type: ignore[operator]
             # TODO(titaiwang): Expose _convert_tensor_to_numpy and _convert_kwargs_for_onnx?
-            input_onnx = [_convert_tensor_to_numpy(x) for x in torch_args]
+            input_onnx = [
+                onnx_proto_utils._convert_tensor_to_numpy(x) for x in torch_args
+            ]
             # deal with dtype and device
```

Comment:
Is this outdated? I think below handles more than these two.

Reply:
Yes, I changed the function name to `_filter_incompatible_and_dtype_convert_kwargs` which should cover 100% what we want in this particular comment.